src/sqlfluff/dialects/dialect_postgres.py

```diff
@@ -3416,3 +3417,82 @@ class TruncateStatementSegment(BaseSegment):
             optional=True,
         ),
     )
+
+
+@postgres_dialect.segment()
+class CopyStatementSegment(BaseSegment):
+    """A `COPY` statement.
+
+    As Specified in https://www.postgresql.org/docs/14/sql-copy.html
+    """
+
+    type = "copy_statement"
+
+    _table_definition = Sequence(
+        Ref("TableReferenceSegment"),
+        Bracketed(Delimited(Ref("ColumnReferenceSegment")), optional=True),
+    )
+
+    _option = match_grammar = Sequence(
+        Ref.keyword("WITH", optional=True),
+        Bracketed(
+            Delimited(
+                AnyNumberOf(
+                    Sequence("FORMAT", Ref("SingleIdentifierGrammar")),
+                    Sequence("FREEZE", Ref("BooleanLiteralGrammar", optional=True)),
+                    Sequence("DELIMITER", Ref("QuotedLiteralSegment")),
+                    Sequence("NULL", Ref("QuotedLiteralSegment")),
+                    Sequence("HEADER", Ref("BooleanLiteralGrammar", optional=True)),
+                    Sequence("QUOTE", Ref("QuotedLiteralSegment")),
+                    Sequence("ESCAPE", Ref("QuotedLiteralSegment")),
+                    Sequence(
+                        "FORCE_QUOTE",
+                        OneOf(
+                            Bracketed(Delimited(Ref("ColumnReferenceSegment"))),
+                            Ref("StarSegment"),
+                        ),
+                    ),
+                    Sequence(
+                        "FORCE_NOT_NULL",
+                        Bracketed(Delimited(Ref("ColumnReferenceSegment"))),
+                    ),
+                    Sequence(
+                        "FORCE_NULL",
+                        Bracketed(Delimited(Ref("ColumnReferenceSegment"))),
+                    ),
+                    Sequence("ENCODING", Ref("QuotedLiteralSegment")),
+                    min_times=1,
+                )
+            )
+        ),
+        optional=True,
+    )
+
+    match_grammar = Sequence(
+        "COPY",
+        OneOf(
+            Sequence(
+                _table_definition,
+                Ref.keyword("FROM"),
```

Comment:
Only need `Ref.keyword` if adding parameters (like optional):

```suggestion
                "FROM",
```